tests/test_pytorch_dataloader_api.py

```diff
@@ -0,0 +1,212 @@
+"""Tests for TileDB integration with Tensorflow Data API."""
+
+import torch
+import tiledb
+import numpy as np
+import unittest
+import tempfile
+
+import torch.nn as nn
+import torch.optim as optim
+
+from tiledb.ml.data_apis.pytorch import PyTorchTileDBDenseDataset
+
+# Test parameters
+NUM_OF_CLASSES = 5
+BATCH_SIZE = 20
+ROWS = 1000
+
+# We test for 2d, 3d, 4d and 5d data
+INPUT_SHAPES = [(10,), (10, 3), (10, 10, 3), (10, 10, 10, 3)]
+
+# We test for single and multiple workers
+NUM_OF_WORKERS = [1, 2, 3, 4, 5]
+
+
+def ingest_in_tiledb(data: np.array, batch_size: int, uri: str):
+    dims = [
+        tiledb.Dim(
+            name="dim_" + str(dim),
+            domain=(0, data.shape[dim] - 1),
+            tile=data.shape[dim] if dim > 0 else batch_size,
+            dtype=np.int32,
+        )
+        for dim in range(data.ndim)
+    ]
+
+    # TileDB schema
+    schema = tiledb.ArraySchema(
+        domain=tiledb.Domain(*dims),
+        sparse=False,
+        attrs=[tiledb.Attr(name="features", dtype=np.float32)],
+    )
+    # Create array
+    tiledb.Array.create(uri, schema)
+
+    # Ingest
+    with tiledb.open(uri, "w") as tiledb_array:
+        tiledb_array[:] = {"features": data}
+
+
+class Net(nn.Module):
+    def __init__(self, shape):
+        super(Net, self).__init__()
+        self.flatten = nn.Flatten()
+        self.linear_relu_stack = nn.Sequential(
+            nn.Linear(np.product(shape), 32),
+            nn.ReLU(),
+            nn.Linear(32, 32),
+            nn.ReLU(),
+            nn.Linear(32, NUM_OF_CLASSES),
+            nn.ReLU(),
+        )
+
+    def forward(self, x):
+        x = self.flatten(x)
+        logits = self.linear_relu_stack(x)
+        return logits
+
+
+class TestTileDBDensePyTorchDataloaderAPI(unittest.TestCase):
+    def test_tiledb_pytorch_data_api_train_with_multiple_dim_data(self):
+        for input_shape in INPUT_SHAPES:
+            for workers in NUM_OF_WORKERS:
+                with self.subTest():
+                    with tempfile.TemporaryDirectory() as tiledb_uri_x, tempfile.TemporaryDirectory() as tiledb_uri_y:
+
+                        dataset_shape_x = (ROWS,) + input_shape
+                        dataset_shape_y = (ROWS,)
+
+                        ingest_in_tiledb(
+                            uri=tiledb_uri_x,
+                            data=np.random.rand(*dataset_shape_x),
+                            batch_size=BATCH_SIZE,
+                        )
+                        ingest_in_tiledb(
+                            uri=tiledb_uri_y,
+                            data=np.random.randint(
+                                low=0, high=NUM_OF_CLASSES, size=dataset_shape_y
+                            ),
+                            batch_size=BATCH_SIZE,
+                        )
+
+                        with tiledb.DenseArray(
```

Comment:
`tiledb.open` is the encouraged way to open dense and sparse arrays. Also `mode=="r"` by default.